toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,249 @@
+	err = updateMarinerCfgWithInitramfs(imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildDracutModule(dracutModuleName string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	// This function will be run inside the chroot to list kernel files in /boot
+	listKernels := func() ([]string, error) {
+		var kernels []string
+		// Assuming the vmlinuz files are located in /boot
+		files, err := filepath.Glob("/boot/vmlinuz-*")
+		if err != nil {
+			return nil, err
+		}
+		for _, file := range files {
+			kernels = append(kernels, filepath.Base(file))
+		}
+		return kernels, nil
+	}
+
+	var kernelFiles []string
+	err = imageChroot.Run(func() error {
+		kernelFiles, err = listKernels()
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list kernels in chroot: %w", err)
+	}
+
+	if len(kernelFiles) == 0 {
+		return fmt.Errorf("no kernels found in chroot environment")
+	}
+
+	// Extract the version from the kernel filename (e.g., vmlinuz-5.15.131.1-2.cm2 -> 5.15.131.1-2.cm2)
+	kernelVersion := strings.TrimPrefix(kernelFiles[0], "vmlinuz-")
+
+	err = imageChroot.Run(func() error {
+		// TODO: Config Dracut module systemd-veritysetup.
+		err = shell.ExecuteLiveWithErr(1, "dracut", "-f", "--kver", kernelVersion, "-a", dracutModuleName)
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("failed to build dracut module - (%s):\n%w", dracutModuleName, err)
+	}
+
+	return nil
+}
+
+func updateMarinerCfgWithInitramfs(imageChroot *safechroot.Chroot) error {
+	var err error
+
+	initramfsPath := filepath.Join(imageChroot.RootDir(), "boot/initramfs-*")
+	// Fetch the initramfs file name.
+	var initramfsFiles []string
+	initramfsFiles, err = filepath.Glob(initramfsPath)
+	if err != nil {
+		return fmt.Errorf("failed to list initramfs file: %w", err)
+	}
+
+	// Ensure an initramfs file is found
+	if len(initramfsFiles) != 1 {
+		return fmt.Errorf("expected one initramfs file, but found %d", len(initramfsFiles))
+	}
+
+	newInitramfs := filepath.Base(initramfsFiles[0])
+
+	cfgPath := filepath.Join(imageChroot.RootDir(), "boot/mariner.cfg")
+	// Update mariner.cfg to reference the new initramfs
+	input, err := os.ReadFile(cfgPath)
+	if err != nil {
+		return fmt.Errorf("failed to read mariner.cfg: %w", err)
+	}
+
+	lines := strings.Split(string(input), "\n")
+	for i, line := range lines {
+		if strings.HasPrefix(line, "mariner_initrd=") {
+			lines[i] = "mariner_initrd=" + newInitramfs
+		}
+	}
+	output := strings.Join(lines, "\n")
+	os.WriteFile(cfgPath, []byte(output), 0644)
+
+	return nil
+}
+
+func updateGrubConfig(dataPartitionIdType imagecustomizerapi.IdType, dataPartitionId string, hashPartitionIdType imagecustomizerapi.IdType, hashPartitionId string, rootHash string, grubCfgFullPath string) error {
+	var err error
+
+	const cmdlineTemplate = "rd.systemd.verity=1 roothash=%s systemd.verity_root_data=%s systemd.verity_root_hash=%s systemd.verity_root_options=panic-on-corruption"
```

Comment:
nit: Place the string literal directly in the call to `fmt.Sprintf`. Splitting the format string from its args makes the code harder to read.